pkg/conf/mesos_target_conf.go

```
@@ -0,0 +1,225 @@
+func CreateMesosTargetConf(targetType string, accountValues []*proto.AccountValue) (probe.TurboTargetConf, error) {
+	var mesosMasterType MesosMasterType
+	if targetType == string(Apache) {
+		mesosMasterType = Apache
+	} else if targetType == string(DCOS) {
+		mesosMasterType = DCOS
+	} else {
+		glog.Errorf("Unknown Mesos Master Type " , targetType)
+		return nil, fmt.Errorf("Unknown Mesos Master Type " , targetType)
+	}
+	config := &MesosTargetConf{
+		Master: mesosMasterType,
+	}
+	for _, accVal := range accountValues {
+		if *accVal.Key ==  string(MasterIP) {
+			config.MasterIP = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(MasterPort) {
+			config.MasterPort = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(MasterUsername) {
+			config.MasterUsername = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(MasterPassword) {
+			config.MasterPassword = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(FrameworkIP) {
+			config.FrameworkIP = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(FrameworkPort) {
+			config.FrameworkPort = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(FrameworkUsername) {
+			config.FrameworkUser = *accVal.StringValue
+		}
+		if *accVal.Key ==  string(FrameworkPassword) {
+			config.FrameworkPassword = *accVal.StringValue
+		}
+	}
+
+	// Provide framework ip for dcos mesos
+	dcosMesosTargetConf(config)
+	// Validate
+	ok, err := config.validate()
+	if !ok {
+		glog.Errorf("[MesosTargetConf] Invalid config : " + err.Error())
+		return nil, fmt.Errorf("[MesosTargetConf] Invalid config : " + err.Error())
+	}
+	return config, nil
+}
+
+
+// Get the Account Values to create VMTTarget in the turbo server corresponding to this client
+func (mesosConf *MesosTargetConf) GetAccountValues() []*proto.AccountValue {
+	if _, err := mesosConf.validate(); err != nil {
+		glog.Infof("[GetAccountValues] mesos target config is not valid %s", err)
+		return nil
```

do we need to return the error as well?

---

Will handle this change in the turbo-go-sdk first